[core]/es_extended/server/onesync.lua

```diff
@@ -7,15 +7,21 @@ ESX.OneSync = {}
 local function getNearbyPlayers(source, closest, distance, ignore)
 	local result = {}
 	local count = 0
+    local playerPed
+    local playerCoords
 	if not distance then distance = 100 end
 	if type(source) == 'number' then
-		source = GetPlayerPed(source)
-
+		playerPed = GetPlayerPed(source)
+        
 		if not source then
-			error("Received invalid first argument (source); should be playerId or vector3 coordinates")
+			error("Received invalid first argument (source); should be playerId")
 		end
 
-		source = GetEntityCoords(source)
+		playerCoords = GetEntityCoords(playerPed)
+
+        if not playerCoords then
+            error("Received nil value (playerCoords); perhaps source is nil at first place?")
```

Comment:
This is not enough, maybe you should return the result, the currently empty table, to stop the code from running.